src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -554,7 +552,11 @@ private string AnalyzeTargetAndCreateUpdateBulkCommand(BulkCopySimpleResultSet i
 
             // Keep track of any result columns that we don't have a local
             // mapping for.
+            #if NETFRAMEWORK
             HashSet<string> unmatchedColumns = new();
+            #else
+            HashSet<string> unmatchedColumns = new(_localColumnMappings.Count);
```

Comment:
Believe it or not, netfx doesn't support a constructor with specifying the size.